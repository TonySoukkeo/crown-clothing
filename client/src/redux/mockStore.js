export const mockCartItem = {
  imageUrl: "http://animage.com",
  price: 30,
  name: "Yellow hat",
  quantity: 10,
};
