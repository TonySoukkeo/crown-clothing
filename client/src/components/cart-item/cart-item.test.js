import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mockCartItem } from "../../redux/mockStore";

configure({
  adapter: new Adapter(),
});

import CartItem from "./cart-item.component";

describe("CartItem component", () => {
  let wrapper;
  let mockItem;

  beforeEach(() => {
    mockItem = {
      imageUrl: "www.image.com",
      price: 35,
      name: "blue hat",
      quantity: 3,
    };

    const mockProps = {
      item: mockItem,
    };

    wrapper = shallow(<CartItem {...mockProps} />);
  });

  it("renders CartItem without crashing", () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("has imageUrl as a src attribute", () => {
    const cartItemImageSrc = wrapper.find("CartItemImage").prop("src");

    expect.assertions(1);
    expect(cartItemImageSrc).toEqual(mockItem.imageUrl);
  });

  it("renders out item name", () => {
    const itemName = wrapper.find("#name").text();

    expect.assertions(1);
    expect(itemName).toEqual(mockItem.name);
  });

  it("renders out item price correctly", () => {
    const itemPrice = +wrapper.find("#price").text();

    expect.assertions(1);
    expect(itemPrice).toEqual(mockItem.price);
  });

  it("renders out item quantity correctly", () => {
    const itemQty = +wrapper.find("#qty").text();

    expect.assertions(1);
    expect(itemQty).toEqual(mockItem.quantity);
  });
});
