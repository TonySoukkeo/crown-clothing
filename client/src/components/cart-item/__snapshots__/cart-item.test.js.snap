// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component renders CartItem without crashing 1`] = `ShallowWrapper {}`;
